File: codility2-4.js
function solution(n, a) {
    // write your code in JavaScript (Node.js 0.12)
    var i=0;
    var l=a.length;
    var arr = Array.apply(null, Array(n)).map(function() { return 0 });
    var max=0;
    for(i; i<l; i++){   
         var k = a[i];
         
         if(k<=n){
            arr[k-1]+=1;
            if(max<arr[k-1])
                max=arr[k-1];
            
         }else{
          //arr = arr.map(function(){return max});
          for(j=0; j<arr.length; j++){
            arr[j] = max;   
          }
         }    
    }
    return arr;   
}

File: codility2-4.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(n, a)
    # write your code in Ruby 2.2
    
    arr = Array.new(n, 0)
    max=0
    a.each_with_index{ |el, i|
       if (el<n)
           arr[el-1]+=1
           if(max<arr[el-1])
             max=arr[el-1]
           end
       else
           arr = Array.new(n, max)
       end
       
        
        
    }
    
    arr
    
end